packages/guides/src/Compiler/NodeTransformers/MenuNodeAddEntryTransformer.php

```diff
@@ -184,4 +172,55 @@ public static function isAbsoluteFile(string $expectedFile): bool
     {
         return str_starts_with($expectedFile, '/');
     }
+
+    /** @param DocumentEntryNode[] $documentEntriesInTree */
+    private function attachDocumentEntriesToParents(
+        array $documentEntriesInTree,
+        CompilerContext $compilerContext,
+        string $currentPath,
+    ): void {
+        foreach ($documentEntriesInTree as $documentEntryInToc) {
+            if ($documentEntryInToc->isRoot() || $currentPath === $documentEntryInToc->getFile()) {
+                // The root page may not be attached to any other
+                continue;
+            }
+
+            if ($documentEntryInToc->getParent() !== null && $documentEntryInToc->getParent() !== $compilerContext->getDocumentNode()->getDocumentEntry()) {
+                $this->logger->warning(sprintf(
```

Comment:
Why is this a warning? Is this an issue somehow that should be fixed? If so, please extend the log message with extra information so the users are aware what is wrong.

Reply:
The warning was not added here but moved in refactoring. great idea anyway